src/utils/global/tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+
+import styles from "./tooltip.module.scss";
+
+interface TooltipProps {
+  text: string;
+  children: React.ReactNode;
+  isShown: boolean;
+  isOption?: boolean;
+  offsetLeft?: number;
+  offsetTop?: number;
+  width?: number;
+}
+
+const Tooltip: React.FC<TooltipProps> = (props) => {
+  const tooltipStyle: React.CSSProperties = {
+    opacity: props.isShown ? 1 : 0,
+    width: `${props.width}rem` || "auto",
+  };
+
+  const tooltipClass = props.isOption
+    ? styles.optionStyles
+    : styles.defaultStyles;
+
+  return (
+    <>
```

Comment:
I would have this wrap the tooltip and its children in a div -- that way, this component can handle the hover detection, so that it becomes an isolated component and you aren't polluting everywhere that uses it with hover detection logic